src/contracts.nit

```diff
@@ -484,19 +548,90 @@ class MEnsure
 	end
 end
 
+# A invariant contract representation
+# This method check if the Object is in a valid state.
+class MInvariant
+	super BottomMContract
+
+	redef fun no_intro_contract(v: ContractsVisitor, a: Array[AAnnotation])
+	do
+		v.toolcontext.warning(a.first.location,"","Useless contract: No contract defined at the introduction of the method")
```

Comment:
you need to add a short unique warning tag, it is used in command line to disable of activate specific warnings